### PR TITLE
Improve test coverage for errors

### PR DESCRIPTION
**Pull request by betasve**
Signed-off-by: Svetoslav Blyahov <svetlio.blyahoff@gmail.com>

Improve configtxgen test coverage (for error messages) as @wlahti advised in my previous PR.

#### Type of change
- Test update

#### Description

Start expecting exact error messages.

### CODE REVIEW

**Review by wlahti — CHANGES_REQUESTED** [review]
Thanks for taking the initiative to improve the tests here. Much appreciated! 

One important requested change (the regexp for the path doesn't work on all operating systems) and a few nits.

**Comment by betasve** [comment]
Pushed some changes I think (as far as I understood your points) are addressing the concerns. Please, let me know if I missed something or something else needs improvement.

**Comment by wlahti** [comment]
Sorry, maybe should've been clearer in the OS-specific nature of the test that fails locally. The temp directory is different depending on the developer's OS, so we need to generalize the regular expression to not care about the beginning of the path. For reference, here's the output when running that test on my mac: 

`
=== RUN   TestUnsuccessfulChannelTxFileCreation
2020-04-14 16:30:43.549 EDT [common.tools.configtxgen.localconfig] Load -> INFO 01b Loaded configuration: /Users/wlahti/gopath/src/github.com/hyperledger/fabric/sampleconfig/configtx.yaml

2020-04-14 16:30:43.549 EDT [common.tools.configtxgen] doOutputChannelCreateTx -> INFO 01c Generating new channel configtx

2020-04-14 16:30:43.551 EDT [common.tools.configtxgen] doOutputChannelCreateTx -> INFO 01d Writing new channel tx

2020-04-14 16:30:43.551 EDT [common.tools.configtxgen] doOutputChannelCreateTx -> INFO 01e Generating new channel configtx

2020-04-14 16:30:43.553 EDT [common.tools.configtxgen] doOutputChannelCreateTx -> INFO 01f Writing new channel tx
    TestUnsuccessfulChannelTxFileCreation: main_test.go:92: 
        	Error Trace:	main_test.go:92
        	Error:      	Expect "Error writing channel create tx: open /var/folders/gy/jb454xb90513cqzgvv7111hr0000gn/T/configtxgen143477252/configtx: permission denied" to match "Error writing channel create tx: open /tmp/configtxgen.+/configtx: permission denied"
        	Test:       	TestUnsuccessfulChannelTxFileCreation
--- FAIL: TestUnsuccessfulChannelTxFileCreation (0.03s)
`

**Review by wlahti — CHANGES_REQUESTED** [review]
Sorry, maybe should've been clearer in the OS-specific nature of the test that fails locally. The temp directory is different depending on the developer's OS, so we need to generalize the regular expression to not care about the beginning of the path. For reference, here's the output when running that test on my mac: 

```
=== RUN   TestUnsuccessfulChannelTxFileCreation
2020-04-14 16:30:43.549 EDT [common.tools.configtxgen.localconfig] Load -> INFO 01b Loaded configuration: /Users/wlahti/gopath/src/github.com/hyperledger/fabric/sampleconfig/configtx.yaml
2020-04-14 16:30:43.549 EDT [common.tools.configtxgen] doOutputChannelCreateTx -> INFO 01c Generating new channel configtx
2020-04-14 16:30:43.551 EDT [common.tools.configtxgen] doOutputChannelCreateTx -> INFO 01d Writing new channel tx
2020-04-14 16:30:43.551 EDT [common.tools.configtxgen] doOutputChannelCreateTx -> INFO 01e Generating new channel configtx
2020-04-14 16:30:43.553 EDT [common.tools.configtxgen] doOutputChannelCreateTx -> INFO 01f Writing new channel tx
    TestUnsuccessfulChannelTxFileCreation: main_test.go:92: 
        	Error Trace:	main_test.go:92
        	Error:      	Expect "Error writing channel create tx: open /var/folders/gy/jb454xb90513cqzgvv7111hr0000gn/T/configtxgen143477252/configtx: permission denied" to match "Error writing channel create tx: open /tmp/configtxgen.+/configtx: permission denied"
        	Test:       	TestUnsuccessfulChannelTxFileCreation
--- FAIL: TestUnsuccessfulChannelTxFileCreation (0.03s)
```

**Comment by betasve** [comment]
Yes, sorry, I didn't really understand your point but it totally makes sense. I should have thought of that. Pushed an update.

File: cmd/configtxgen/main_test.go
Comment (wlahti) [inline cmd/configtxgen/main_test.go]:
```suggestion
	assert.EqualError(t, doOutputChannelCreateTx(config, nil, "foo", configTxDest), fmt.Sprintf("Error writing channel create tx: open %s: permission denied", configTxDest))
```
Not trying to stall on this one, but considering I'm not a maintainer and can't merge this in anyways, can't we simplify this down to just one line and avoid the regular expression by formatting in the path since we have the variable with the exact path. Tried this out locally and it works for me. Seems more flexible to me in case somebody down the line decided to change the name of the tmpDir.

Comment (betasve) [reply cmd/configtxgen/main_test.go]: Sure, sure, I totally got your point. Pushing a change now. Let me know about the new version.

**Review by wlahti — DISMISSED** [review]
Looks good to me now, thanks!

File: cmd/configtxgen/main_test.go
Comment (caod123) [inline cmd/configtxgen/main_test.go]: I know you didn't introduce these error messages, but per the general Go programming conventions, error strings should not start capitalized https://github.com/golang/go/wiki/CodeReviewComments#error-strings would you be able to go through the production code as well as the tests in this path and update the error strings accordingly? There's a lot of places in Fabric where we don't properly follow this convention but might as well start here since we're touching this path.

File: cmd/configtxgen/main_test.go
Comment (caod123) [inline cmd/configtxgen/main_test.go]: Same here

File: cmd/configtxgen/main_test.go
Comment (caod123) [inline cmd/configtxgen/main_test.go:63]: Interestingly enough we follow the convention here, further demonstrating how inconsistent error handling is in Fabric

File: cmd/configtxgen/main_test.go
Comment (caod123) [inline cmd/configtxgen/main_test.go]: And here

File: cmd/configtxgen/main_test.go
Comment (caod123) [inline cmd/configtxgen/main_test.go]: And here

File: cmd/configtxgen/main_test.go
Comment (caod123) [inline cmd/configtxgen/main_test.go]: And here

File: cmd/configtxgen/main_test.go
Comment (caod123) [inline cmd/configtxgen/main_test.go]: And here

**Review by caod123 — CHANGES_REQUESTED** [review]
Some general comments on error string conventions

**Review by sykesm — APPROVED** [review]

**Comment by wlahti** [comment]
Can you rebase this on master and we'll see if we can get a maintainer to merge as-is? I think the capitalization of the error messages can be done in a follow-up PR considering they've had that capitalization for years.

**Review by caod123 — APPROVED** [review]

**Comment by caod123** [comment]
Looks like there aren't any conflicts with master, merging as is

**Pull request merged**
